internal/transport/proxy_ext_test.go

```diff
@@ -504,21 +477,15 @@ func (s) TestBasicAuthInNewClientWithProxy(t *testing.T) {
 	}
 	pServer := transporttestutils.HTTPProxy(t, reqCheck, false)
 
-	// Overwrite the function in the test and restore them in defer.
-	hpfe := func(req *http.Request) (*url.URL, error) {
-		if req.URL.Host == unresolvedTargetURI {
-			return &url.URL{
-				Scheme: "https",
-				User:   url.UserPassword(user, password),
-				Host:   pServer.Addr,
-			}, nil
-		}
-		t.Errorf("Unexpected request host to proxy: %s want %s", req.URL.Host, unresolvedTargetURI)
-		return nil, nil
+	t.Setenv("HTTPS_PROXY", user+":"+password+"@"+pServer.Addr)
+
+	origHTTPSProxyFromEnvironment := delegatingresolver.HTTPSProxyFromEnvironment
+	delegatingresolver.HTTPSProxyFromEnvironment = func(req *http.Request) (*url.URL, error) {
+		return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
```

Comment:
nit: For the benefit of future readers, please add a comment mentioning this done to read the env var in every test run.